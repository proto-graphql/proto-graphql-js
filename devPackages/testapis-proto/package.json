{
  "name": "@proto-graphql/testapis-proto",
  "version": "0.1.1",
  "description": "Generated codes from protobuf IDL",
  "type": "module",
  "main": "./dist/index.cjs",
  "module": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": {
        "types": "./dist/index.d.ts",
        "default": "./dist/index.js"
      },
      "require": {
        "types": "./dist/index.d.ts",
        "default": "./dist/index.cjs"
      }
    }
  },
  "sideEffects": false,
  "private": true,
  "scripts": {
    "build": "tsup",
    "prebuild": "./scripts/build-proto.ts"
  },
  "dependencies": {
    "@bufbuild/protobuf": "catalog:"
  },
  "devDependencies": {
    "@proto-graphql/tsconfig": "workspace:*",
    "glob": "^10.4.5"
  }
}
