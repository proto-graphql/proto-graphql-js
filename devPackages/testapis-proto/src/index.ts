import { create, fromBinary } from "@bufbuild/protobuf";
import {
  type CodeGeneratorRequest,
  CodeGeneratorRequestSchema,
  type FileDescriptorSet,
  FileDescriptorSetSchema,
} from "@bufbuild/protobuf/wkt";
import { fileDescriptorSetBins } from "./__generated__/fileDescriptorSetBins";

function objectKeys<K extends string>(obj: Record<K, unknown>): K[] {
  return Object.keys(obj) as K[];
}

export const testapisPackages = objectKeys(fileDescriptorSetBins);
export type TestapisPackage = keyof typeof fileDescriptorSetBins;

export function getTestapisFileDescriptorSet(
  pkg: TestapisPackage,
): FileDescriptorSet {
  return fromBinary(
    FileDescriptorSetSchema,
    Buffer.from(fileDescriptorSetBins[pkg], "base64"),
  );
}

export function buildCodeGeneratorRequest(
  pkg: TestapisPackage,
  { param }: { param?: string } = {},
): CodeGeneratorRequest {
  const descSet = getTestapisFileDescriptorSet(pkg);
  const req = create(CodeGeneratorRequestSchema, { parameter: param });

  for (const fd of descSet.file) {
    req.protoFile.push(fd);

    const filename = fd.name;
    const pat = new RegExp(`^${pkg.replace(/\./g, "/")}/[^/]+\\.proto$`);
    if (filename && pat.test(filename)) {
      req.fileToGenerate.push(filename);
    }
  }

  return req;
}
