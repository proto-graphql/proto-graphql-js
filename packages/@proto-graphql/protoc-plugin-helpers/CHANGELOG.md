# @proto-graphql/protoc-plugin-helpers

## 0.4.2

### Patch Changes

- [#417](https://github.com/proto-graphql/proto-graphql-js/pull/417) [`2ae48d1`](https://github.com/proto-graphql/proto-graphql-js/commit/2ae48d1576a15a47f3f1083fb78eec69507d601a) Thanks [@izumin5210](https://github.com/izumin5210)! - use protobuf-es@v2 internally

- [#427](https://github.com/proto-graphql/proto-graphql-js/pull/427) [`c681582`](https://github.com/proto-graphql/proto-graphql-js/commit/c681582431637946cebdaa82b9559bb5ab52b760) Thanks [@izumin5210](https://github.com/izumin5210)! - refactor: parse options with protoplugin api

- [#428](https://github.com/proto-graphql/proto-graphql-js/pull/428) [`e3cee38`](https://github.com/proto-graphql/proto-graphql-js/commit/e3cee38438fbb4c0fbd177566aa75e83e11221ed) Thanks [@izumin5210](https://github.com/izumin5210)! - refactor: migrate protoc plugins to ESM

- Updated dependencies [[`2ae48d1`](https://github.com/proto-graphql/proto-graphql-js/commit/2ae48d1576a15a47f3f1083fb78eec69507d601a), [`e3cee38`](https://github.com/proto-graphql/proto-graphql-js/commit/e3cee38438fbb4c0fbd177566aa75e83e11221ed)]:
  - @proto-graphql/codegen-core@0.5.2

## 0.4.1

### Patch Changes

- [#371](https://github.com/proto-graphql/proto-graphql-js/pull/371) [`d8be975`](https://github.com/proto-graphql/proto-graphql-js/commit/d8be975c7e716e11cbfc662ab7518020525b75d7) Thanks [@izumin5210](https://github.com/izumin5210)! - build: set `pulic` to `publishConfig.access`

- Updated dependencies [[`d8be975`](https://github.com/proto-graphql/proto-graphql-js/commit/d8be975c7e716e11cbfc662ab7518020525b75d7)]:
  - @proto-graphql/codegen-core@0.5.1

## 0.4.0

### Minor Changes

- [#369](https://github.com/proto-graphql/proto-graphql-js/pull/369) [`967ed0e`](https://github.com/proto-graphql/proto-graphql-js/commit/967ed0e5a5a9d8dea7e7ef46a220cc57048844da) Thanks [@izumin5210](https://github.com/izumin5210)! - support protobuf-es@v1 on protoc-gen-pothos

### Patch Changes

- Updated dependencies [[`967ed0e`](https://github.com/proto-graphql/proto-graphql-js/commit/967ed0e5a5a9d8dea7e7ef46a220cc57048844da)]:
  - @proto-graphql/codegen-core@0.5.0

## 0.3.0

### Minor Changes

- [#327](https://github.com/proto-graphql/proto-graphql-js/pull/327) [`cb38fa7`](https://github.com/proto-graphql/proto-graphql-js/commit/cb38fa72abf3f1ff2fba0fcfc75d725f63c0ff85) Thanks [@izumin5210](https://github.com/izumin5210)! - support `ignore_non_message_oneof_fields` option

- [#346](https://github.com/proto-graphql/proto-graphql-js/pull/346) [`fa1e455`](https://github.com/proto-graphql/proto-graphql-js/commit/fa1e455561e2ea2a80accf7e80e2ff731e80eac6) Thanks [@izumin5210](https://github.com/izumin5210)! - refactor: use `@bufbuild/protoplugin`

- [#310](https://github.com/proto-graphql/proto-graphql-js/pull/310) [`8f7ad44`](https://github.com/proto-graphql/proto-graphql-js/commit/8f7ad44e56356867a3f95c0d08437d31bc4b8f33) Thanks [@izumin5210](https://github.com/izumin5210)! - support scalar option for define custom scalar type mappings instead of legacy option

  BREAKING CHANGE: drop `long_number` and `custom_type` options

  #### Migration

  `long_number` and `custom_type` can be replaced with `scalar`.

  ```patch
   plugins:
     name: pothos
     out: src/__generated__/schema
     opt:
  -    - long_number=Int
  +    - scalar=int64=Int
  +    - scalar=uint64=Int
  +    - scalar=sint64=Int
  +    - scalar=fixed64=Int
  +    - scalar=sfixed64=Int
  +    - scalar=google.protobuf.Int64Value=Int
  +    - scalar=google.protobuf.UInt64Value=Int
  +    - scalar=google.protobuf.SInt64Value=Int
  +    - scalar=google.protobuf.Fixed64Value=Int
  +    - scalar=google.protobuf.SFixed64Value=Int
  ```

  ```patch
   plugins:
     name: pothos
     out: src/__generated__/schema
     opt:
  -    - custom_type=google.type.Date=Date
  +    - scalar=google.type.Date=Date
  ```

### Patch Changes

- [#354](https://github.com/proto-graphql/proto-graphql-js/pull/354) [`0b532bc`](https://github.com/proto-graphql/proto-graphql-js/commit/0b532bcc1fb21d2364a911d2234bd30449804cb6) Thanks [@izumin5210](https://github.com/izumin5210)! - build with tsup

- Updated dependencies [[`9fd6869`](https://github.com/proto-graphql/proto-graphql-js/commit/9fd6869ead97335506076bcb7086a9d561aa4bb9), [`0b532bc`](https://github.com/proto-graphql/proto-graphql-js/commit/0b532bcc1fb21d2364a911d2234bd30449804cb6), [`cb38fa7`](https://github.com/proto-graphql/proto-graphql-js/commit/cb38fa72abf3f1ff2fba0fcfc75d725f63c0ff85), [`2b3d87f`](https://github.com/proto-graphql/proto-graphql-js/commit/2b3d87fcc05b3ef6ae7921ef8e72a667ec751159), [`fa1e455`](https://github.com/proto-graphql/proto-graphql-js/commit/fa1e455561e2ea2a80accf7e80e2ff731e80eac6), [`8f7ad44`](https://github.com/proto-graphql/proto-graphql-js/commit/8f7ad44e56356867a3f95c0d08437d31bc4b8f33), [`1166d7b`](https://github.com/proto-graphql/proto-graphql-js/commit/1166d7bb4e183e06e7964be897a204d864b94381)]:
  - @proto-graphql/codegen-core@0.4.0

## 0.2.7

### Patch Changes

- Updated dependencies [[`16b1fac`](https://github.com/proto-graphql/proto-graphql-js/commit/16b1facf48cae5ceb959bf3e81643b7fc58011fc)]:
  - @proto-graphql/codegen-core@0.3.5

## 0.2.6

### Patch Changes

- [#308](https://github.com/proto-graphql/proto-graphql-js/pull/308) [`5a650c6`](https://github.com/proto-graphql/proto-graphql-js/commit/5a650c6ace5f6132a7dcfd76be8c2c45d84611ab) Thanks [@izumin5210](https://github.com/izumin5210)! - redo the last release

- Updated dependencies [[`5a650c6`](https://github.com/proto-graphql/proto-graphql-js/commit/5a650c6ace5f6132a7dcfd76be8c2c45d84611ab)]:
  - @proto-graphql/codegen-core@0.3.4
  - @proto-graphql/proto-descriptors@0.3.3

## 0.2.5

### Patch Changes

- [#294](https://github.com/proto-graphql/proto-graphql-js/pull/294) [`709c7da`](https://github.com/proto-graphql/proto-graphql-js/commit/709c7da021fb503efeaa7ec4a3485fd166204563) Thanks [@izumin5210](https://github.com/izumin5210)! - refer internal dependencies with `workspace:`

- Updated dependencies [[`709c7da`](https://github.com/proto-graphql/proto-graphql-js/commit/709c7da021fb503efeaa7ec4a3485fd166204563), [`fc4512a`](https://github.com/proto-graphql/proto-graphql-js/commit/fc4512a1db42b7bbc4683e6ad4eb9fbf3a6d24f0)]:
  - @proto-graphql/codegen-core@0.3.3
  - @proto-graphql/proto-descriptors@0.3.2

## 0.2.4

### Patch Changes

- [#290](https://github.com/proto-graphql/proto-graphql-js/pull/290) [`ff3638b`](https://github.com/proto-graphql/proto-graphql-js/commit/ff3638b382953cef02df774df0e93fbd991548ee) Thanks [@izumin5210](https://github.com/izumin5210)! - add PROTO3_OPTIONAL to supported features

## 0.2.3

### Patch Changes

- [#253](https://github.com/proto-graphql/proto-graphql-js/pull/253) [`db7fcec`](https://github.com/proto-graphql/proto-graphql-js/commit/db7fcec7aa987037523388f9fe4ae9b8c3b4c6a4) Thanks [@izumin5210](https://github.com/izumin5210)! - support `emit_imported_files` option

- Updated dependencies [[`db7fcec`](https://github.com/proto-graphql/proto-graphql-js/commit/db7fcec7aa987037523388f9fe4ae9b8c3b4c6a4), [`70295da`](https://github.com/proto-graphql/proto-graphql-js/commit/70295dae5e83a5ee4ee5185e670348c1a88979e2)]:
  - @proto-graphql/codegen-core@0.3.2
  - @proto-graphql/proto-descriptors@0.3.1

## 0.2.2

### Patch Changes

- [#243](https://github.com/proto-graphql/proto-graphql-js/pull/243) [`3aa729a`](https://github.com/proto-graphql/proto-graphql-js/commit/3aa729a62a36aa782e8c2153bb3b789e9551ae00) Thanks [@izumin5210](https://github.com/izumin5210)! - remove unused code from codegen-core

- Updated dependencies [[`3aa729a`](https://github.com/proto-graphql/proto-graphql-js/commit/3aa729a62a36aa782e8c2153bb3b789e9551ae00), [`cef3d3e`](https://github.com/proto-graphql/proto-graphql-js/commit/cef3d3e512e616f98869c12060d592c3ff887aa9), [`f072608`](https://github.com/proto-graphql/proto-graphql-js/commit/f0726082c9ad728ff5ad648e79ab7862f85a1a97)]:
  - @proto-graphql/codegen-core@0.3.0
  - @proto-graphql/proto-descriptors@0.3.0

## 0.2.1

### Patch Changes

- Updated dependencies [[`a2a3e60`](https://github.com/proto-graphql/proto-graphql-js/commit/a2a3e60789f75c31c600333afd830a02ac0cccf8)]:
  - @proto-graphql/codegen-core@0.2.2
  - @proto-graphql/proto-descriptors@0.2.0

## 0.2.0

### Minor Changes

- [#224](https://github.com/proto-graphql/proto-graphql-js/pull/224) [`6c25f40`](https://github.com/proto-graphql/proto-graphql-js/commit/6c25f4035f1b788a63bf005dbc52ca5d0bd5f2a2) Thanks [@izumin5210](https://github.com/izumin5210)! - support `long_number` param for mapping 64bit numbers to GraphQL Int

### Patch Changes

- Updated dependencies [[`6c25f40`](https://github.com/proto-graphql/proto-graphql-js/commit/6c25f4035f1b788a63bf005dbc52ca5d0bd5f2a2)]:
  - @proto-graphql/codegen-core@0.2.0

## 0.1.1

### Patch Changes

- [#213](https://github.com/proto-graphql/proto-graphql-js/pull/213) [`c5b988a`](https://github.com/proto-graphql/proto-graphql-js/commit/c5b988a4a1eac5d56d29572b593ae7643e23bd88) Thanks [@izumin5210](https://github.com/izumin5210)! - handle `pothos_builder_path` param correctly

- Updated dependencies [[`ba647eb`](https://github.com/proto-graphql/proto-graphql-js/commit/ba647eb584850fee9c632a76cc1c028ce8ccd725)]:
  - @proto-graphql/codegen-core@0.1.1

## 0.1.0

### Minor Changes

- [#200](https://github.com/proto-graphql/proto-graphql-js/pull/200) [`515eabd`](https://github.com/proto-graphql/proto-graphql-js/commit/515eabd2f39baa0a99ae057b1b30a4ccc4149f66) Thanks [@izumin5210](https://github.com/izumin5210)! - support Pothos GraphQL

### Patch Changes

- Updated dependencies [[`515eabd`](https://github.com/proto-graphql/proto-graphql-js/commit/515eabd2f39baa0a99ae057b1b30a4ccc4149f66)]:
  - @proto-graphql/codegen-core@0.1.0
  - @proto-graphql/proto-descriptors@0.1.0
