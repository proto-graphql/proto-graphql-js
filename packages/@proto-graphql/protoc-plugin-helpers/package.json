{
  "name": "@proto-graphql/protoc-plugin-helpers",
  "version": "0.4.1",
  "description": "",
  "keywords": ["graphql", "grpc", "nexus", "protobuf", "typescript"],
  "repository": "git@github.com:proto-graphql/proto-graphql-js.git",
  "author": "izumin5210 <m@izum.in>",
  "license": "MIT",
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.mjs",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts",
      "default": "./dist/index.js"
    }
  },
  "private": false,
  "publishConfig": {
    "access": "public"
  },
  "engines": {
    "node": ">= 14.0.0"
  },
  "dependencies": {
    "@bufbuild/protobuf": "catalog:",
    "@bufbuild/protoplugin": "catalog:",
    "@proto-graphql/codegen-core": "workspace:*"
  },
  "devDependencies": {
    "@proto-graphql/tsconfig": "workspace:*"
  },
  "files": ["src/", "dist/", "!src/**/*.test.ts", "!src/**/__tests__/"],
  "scripts": {
    "build": "tsup",
    "test": "vitest run --passWithNoTests"
  }
}
