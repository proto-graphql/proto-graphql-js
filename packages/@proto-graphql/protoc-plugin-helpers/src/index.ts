export { createTsGenerator } from "./process.js";
export { parseNexusOptions, parsePothosOptions } from "./options.js";
export type { Options } from "./options.js";
