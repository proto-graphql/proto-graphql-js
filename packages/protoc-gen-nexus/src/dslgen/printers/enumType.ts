import type { Registry } from "@bufbuild/protobuf";
import {
  type EnumType,
  compact,
  protobufGraphQLExtensions,
} from "@proto-graphql/codegen-core";
import { type Code, code, literalOf } from "ts-poet";

import { impNexus, nexusTypeDef } from "./util.js";

/**
 * @example
 * ```ts
 * export cosnt Hello = enumType({
 *   name: "Hello",
 *   // ...
 * })
 * ```
 */
export function createEnumTypeCode(type: EnumType, registry: Registry): Code {
  const typeOpts = {
    name: type.typeName,
    description: type.description,
    members: type.values.map((ev) => ({
      name: ev.name,
      value: ev.number,
      description: ev.description,
      deprecation: ev.deprecationReason,
      extensions: protobufGraphQLExtensions(ev, registry),
    })),
    extensions: protobufGraphQLExtensions(type, registry),
  };
  return code`export const ${nexusTypeDef(type)} = ${impNexus(
    "enumType",
  )}(${literalOf(compact(typeOpts))});`;
}
