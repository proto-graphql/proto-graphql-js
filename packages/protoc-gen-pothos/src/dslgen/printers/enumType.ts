import type { Registry } from "@bufbuild/protobuf";
import {
  type EnumType,
  compact,
  protoType,
  protobufGraphQLExtensions,
} from "@proto-graphql/codegen-core";
import { type Code, code, imp, joinCode, literalOf } from "ts-poet";

import { type PothosPrinterOptions, pothosBuilder, pothosRef } from "./util.js";

/**
 * @example
 * ```ts
 * export cosnt Hello$Ref: EnumRef<Hello, Hello> = builder.enumType("Hello", {
 *   values: [
 *     // ...
 *   ],
 *   // ...
 * })
 * ```
 */
export function createEnumTypeCode(
  type: EnumType,
  registry: Registry,
  opts: PothosPrinterOptions,
): Code {
  const typeOpts = {
    description: type.description,
    values: code`{${joinCode(
      type.values
        .filter((v) => !v.isIgnored() && !v.isUnespecified())
        .map(
          (ev) =>
            code`${ev.name}: ${literalOf(
              compact({
                description: ev.description,
                deprecationReason: ev.deprecationReason,
                value: ev.number,
                extensions: protobufGraphQLExtensions(ev, registry),
              }),
            )},`,
        ),
    )}} as const`,
    extensions: protobufGraphQLExtensions(type, registry),
  };

  const protoTypeExpr = protoType(type.proto, opts);
  // EnumRef<Hello, Hello>
  const refTypeExpr = code`${imp(
    "EnumRef@@pothos/core",
  )}<${protoTypeExpr}, ${protoTypeExpr}>`;

  return code`
    export const ${pothosRef(type)}: ${refTypeExpr} =
      ${pothosBuilder(type, opts)}.enumType(${literalOf(
        type.typeName,
      )}, ${literalOf(compact(typeOpts))});
  `;
}
