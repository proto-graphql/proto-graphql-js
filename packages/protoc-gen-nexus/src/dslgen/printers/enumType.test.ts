import { createEcmaScriptPlugin } from "@bufbuild/protoplugin";
import { buildCodeGeneratorRequest } from "@proto-graphql/testapis-proto";
import { 
  EnumType, 
  collectTypesFromFile,
  createRegistryFromSchema,
  defaultScalarMappingForTsProto,
  defaultScalarMapping
} from "@proto-graphql/codegen-core";
import { createTsGenerator, parseNexusOptions } from "@proto-graphql/protoc-plugin-helpers";
import { code } from "ts-poet";
import { describe, expect, test } from "vitest";
import { createEnumTypeCode } from "./enumType.js";

// Helper to extract EnumType instances and generate code
function generateEnumTypeCode(packageName: string, enumTypeName: string, protobuf: "google-protobuf" | "protobufjs") {
  const req = buildCodeGeneratorRequest(packageName);
  
  // Default type options
  const typeOptions = {
    partialInputs: false,
    scalarMapping: defaultScalarMapping,
    ignoreNonMessageOneofFields: false,
  };
  
  // Create a minimal plugin to get Schema
  const plugin = createEcmaScriptPlugin({
    name: "test-plugin",
    version: "v1.0.0",
    generateTs: createTsGenerator({
      generateFiles: (schema, file) => {
        // Just return, we only need the schema
      },
      dsl: "nexus",
    }),
    parseOptions: parseNexusOptions,
  });
  
  // Run the plugin to get the transformed schema
  const resp = plugin.run(req);
  
  // Now we need to recreate the schema to access its internals
  const testPlugin = createEcmaScriptPlugin({
    name: "test-plugin",
    version: "v1.0.0",
    generateTs: (schema) => {
      const registry = createRegistryFromSchema(schema);
      
      // Find the target file
      const targetFile = schema.allFiles.find(f => f.name.includes(packageName.split('.')[1]));
      if (!targetFile) throw new Error(`File for ${packageName} not found`);
      
      const types = collectTypesFromFile(targetFile, typeOptions, schema.allFiles);
      const enumType = types.find(t => t.typeName === enumTypeName && t instanceof EnumType) as EnumType;
      if (!enumType) throw new Error(`${enumTypeName} type not found`);
      
      const generatedCode = createEnumTypeCode(enumType, registry);
      
      // Store the result for testing
      (global as any).testResult = generatedCode.toString();
    },
    parseOptions: parseNexusOptions,
  });
  
  testPlugin.run(req);
  
  const result = (global as any).testResult;
  delete (global as any).testResult;
  return result;
}

describe("createEnumTypeCode", () => {
  describe("google-protobuf", () => {
    test("generates code for a simple enum", () => {
      const code = generateEnumTypeCode("testapis.enums", "MyEnum", "google-protobuf");
      expect(code).toMatchSnapshot();
    });

    test("generates code for an enum without unspecified", () => {
      const code = generateEnumTypeCode("testapis.enums", "MyEnumWithoutUnspecified", "google-protobuf");
      expect(code).toMatchSnapshot();
    });

    test("generates code for nested enum", () => {
      const code = generateEnumTypeCode("testapis.nested", "ParentMessageNestedEnum", "google-protobuf");
      expect(code).toMatchSnapshot();
    });

    test("generates code for enum with extensions", () => {
      const code = generateEnumTypeCode("testapis.extensions", "TestPrefixPrefixedEnum", "google-protobuf");
      expect(code).toMatchSnapshot();
    });
  });

  describe("protobufjs", () => {
    test("generates code for a simple enum", () => {
      const code = generateEnumTypeCode("testapis.enums", "MyEnum", "protobufjs");
      expect(code).toMatchSnapshot();
    });

    test("generates code for an enum without unspecified", () => {
      const code = generateEnumTypeCode("testapis.enums", "MyEnumWithoutUnspecified", "protobufjs");
      expect(code).toMatchSnapshot();
    });

    test("generates code for nested enum", () => {
      const code = generateEnumTypeCode("testapis.nested", "ParentMessageNestedEnum", "protobufjs");
      expect(code).toMatchSnapshot();
    });
  });
});