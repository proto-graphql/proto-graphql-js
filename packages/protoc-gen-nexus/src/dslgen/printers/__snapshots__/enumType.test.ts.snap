// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createEnumTypeCode > google-protobuf > generates code for a simple enum 1`] = `
"import { enumType } from "nexus";

export const MyEnum = enumType({
  "name": "MyEnum",
  "members": [{ "name": "FOO", "value": 1, "extensions": { "protobufEnumValue": { "name": "MY_ENUM_FOO" } } }, {
    "name": "BAR",
    "value": 2,
    "description": "This is Bar.",
    "extensions": { "protobufEnumValue": { "name": "MY_ENUM_BAR" } },
  }, { "name": "BAZ", "value": 3, "extensions": { "protobufEnumValue": { "name": "MY_ENUM_BAZ" } } }],
  "extensions": {
    "protobufEnum": { "name": "MyEnum", "fullName": "testapi.enums.MyEnum", "package": "testapi.enums" },
  },
});
"
`;

exports[`createEnumTypeCode > google-protobuf > generates code for an enum without unspecified 1`] = `
"import { enumType } from "nexus";

export const MyEnumWithoutUnspecified = enumType({
  "name": "MyEnumWithoutUnspecified",
  "members": [{
    "name": "FOO",
    "value": 0,
    "extensions": { "protobufEnumValue": { "name": "MY_ENUM_WITHOUT_UNSPECIFIED_FOO" } },
  }, {
    "name": "BAR",
    "value": 1,
    "extensions": { "protobufEnumValue": { "name": "MY_ENUM_WITHOUT_UNSPECIFIED_BAR" } },
  }, {
    "name": "BAZ",
    "value": 2,
    "extensions": { "protobufEnumValue": { "name": "MY_ENUM_WITHOUT_UNSPECIFIED_BAZ" } },
  }],
  "extensions": {
    "protobufEnum": {
      "name": "MyEnumWithoutUnspecified",
      "fullName": "testapi.enums.MyEnumWithoutUnspecified",
      "package": "testapi.enums",
    },
  },
});
"
`;

exports[`createEnumTypeCode > google-protobuf > generates code for enum with extensions 1`] = `
"import { enumType } from "nexus";

export const TestPrefixPrefixedEnum = enumType({
  "name": "TestPrefixPrefixedEnum",
  "members": [
    { "name": "PREFIXED_FOO", "value": 1, "extensions": { "protobufEnumValue": { "name": "PREFIXED_FOO" } } },
    { "name": "PREFIXED_BAR", "value": 2, "extensions": { "protobufEnumValue": { "name": "PREFIXED_BAR" } } },
  ],
  "extensions": {
    "protobufEnum": {
      "name": "PrefixedEnum",
      "fullName": "testapis.extensions.PrefixedEnum",
      "package": "testapis.extensions",
    },
  },
});
"
`;

exports[`createEnumTypeCode > google-protobuf > generates code for nested enum 1`] = `
"import { enumType } from "nexus";

export const ParentMessageNestedEnum = enumType({
  "name": "ParentMessageNestedEnum",
  "members": [{ "name": "FOO", "value": 1, "extensions": { "protobufEnumValue": { "name": "FOO" } } }, {
    "name": "BAR",
    "value": 2,
    "extensions": { "protobufEnumValue": { "name": "BAR" } },
  }],
  "extensions": {
    "protobufEnum": {
      "name": "NestedEnum",
      "fullName": "testapis.nested.ParentMessage.NestedEnum",
      "package": "testapis.nested",
    },
  },
});
"
`;

exports[`createEnumTypeCode > protobufjs > generates code for a simple enum 1`] = `
"import { enumType } from "nexus";

export const MyEnum = enumType({
  "name": "MyEnum",
  "members": [{ "name": "FOO", "value": 1, "extensions": { "protobufEnumValue": { "name": "MY_ENUM_FOO" } } }, {
    "name": "BAR",
    "value": 2,
    "description": "This is Bar.",
    "extensions": { "protobufEnumValue": { "name": "MY_ENUM_BAR" } },
  }, { "name": "BAZ", "value": 3, "extensions": { "protobufEnumValue": { "name": "MY_ENUM_BAZ" } } }],
  "extensions": {
    "protobufEnum": { "name": "MyEnum", "fullName": "testapi.enums.MyEnum", "package": "testapi.enums" },
  },
});
"
`;

exports[`createEnumTypeCode > protobufjs > generates code for an enum without unspecified 1`] = `
"import { enumType } from "nexus";

export const MyEnumWithoutUnspecified = enumType({
  "name": "MyEnumWithoutUnspecified",
  "members": [{
    "name": "FOO",
    "value": 0,
    "extensions": { "protobufEnumValue": { "name": "MY_ENUM_WITHOUT_UNSPECIFIED_FOO" } },
  }, {
    "name": "BAR",
    "value": 1,
    "extensions": { "protobufEnumValue": { "name": "MY_ENUM_WITHOUT_UNSPECIFIED_BAR" } },
  }, {
    "name": "BAZ",
    "value": 2,
    "extensions": { "protobufEnumValue": { "name": "MY_ENUM_WITHOUT_UNSPECIFIED_BAZ" } },
  }],
  "extensions": {
    "protobufEnum": {
      "name": "MyEnumWithoutUnspecified",
      "fullName": "testapi.enums.MyEnumWithoutUnspecified",
      "package": "testapi.enums",
    },
  },
});
"
`;

exports[`createEnumTypeCode > protobufjs > generates code for nested enum 1`] = `
"import { enumType } from "nexus";

export const ParentMessageNestedEnum = enumType({
  "name": "ParentMessageNestedEnum",
  "members": [{ "name": "FOO", "value": 1, "extensions": { "protobufEnumValue": { "name": "FOO" } } }, {
    "name": "BAR",
    "value": 2,
    "extensions": { "protobufEnumValue": { "name": "BAR" } },
  }],
  "extensions": {
    "protobufEnum": {
      "name": "NestedEnum",
      "fullName": "testapis.nested.ParentMessage.NestedEnum",
      "package": "testapis.nested",
    },
  },
});
"
`;
